src/pages/previous.js

```diff
@@ -0,0 +1,152 @@
+  };
+
+  async function getAll(questionId) {
+    const answers = await GET("questions/" + questionId + "/answers");
+    await Promise.all(
+      answers.map((answer) => {
+        return new Promise((res) => {
+          GET("users/" + answer["user_id"].toString()).then((user) =>
+            res({ user, body: answer.body, time: new Date(answer.created_at) })
+          );
+        });
+      })
+    ).then((values) => {
+      const data = values.map((value) => {
+        return {
+          email: value.user.email,
+          body: value.body,
+          time: value.time,
+          user: value.user.id,
+        };
+      });
+      setAnswers((prevAnswers) => ({
+        ...prevAnswers,
+        [questionId]: data,
+      }));
+    });
+  }
+
+  function renderDate(date) {
+    const startDate = new Date(date.getTime());
+    const endDate = new Date(date.getTime());
+
+    const daysToSubtract = startDate.getDay() + 7;
+
+    startDate.setDate(startDate.getDate() - daysToSubtract + 1);
+    endDate.setDate(endDate.getDate() - startDate.getDay());
+
+    const startDay = startDate.getDate();
+    const startMonth = startDate.getMonth() + 1;
+
+    const endDay = endDate.getDate();
+    const endMonth = endDate.getMonth() + 1;
+
+    return `${startDay}/${startMonth}-${endDay}/${endMonth}`;
+  }
+
+  return (
+    <div className="bg-previous min-h-screen bg-no-repeat bg-cover">
+      <NavBar />
+      <div className="flex mt-8 ml-32 text-xl">
+        <h1 className="text-cerulean hover:underline">
+          {" "}
+          <a href="#/question">Weekly Question</a>
+        </h1>
+        <h1 className="text-white no-underline"> &nbsp;&gt; History</h1>
+      </div>
+
+      {questions
+        .sort((a, b) => b.time.getTime() - a.time.getTime())
+        .map((questionItem) => (
+          <div
+            className=" mt-16 ml-96 gap-8 w-4/6 overflow-scroll no-scrollbar"
+            key={questionItem.id}
+          >
+            <div className="border-b-2 border-white px-2 pb-8">
+              <div className="flex gap-6">
+                <h1 className="text-white w-28 pb-8">
+                  {renderDate(questionItem.time)}
+                </h1>
+                <h1 className="text-white w-9">
+                  {questionItem.time.getFullYear()}
+                </h1>
+                <h1 className="text-white text-xl w-full">
+                  {questionItem.body}
+                </h1>
+                <img
+                  src={Chevron}
+                  onClick={() => viewAnswers(questionItem.id)}
+                  alt="Chevron icon"
+                  className={`cursor-pointer ${
+                    answersVisibility[questionItem.id] ? "rotate-180" : ""
+                  }`}
+                />
+              </div>
+              <div className="items-center overflow-scroll no-scrollbar max-h-96">
+                {answersVisibility[questionItem.id] &&
+                  answers[questionItem.id]?.map((item) => (
```

Comment:
```suggestion
                  answers[questionItem.id]?.map((answer) => (
```

Let's focus more on better names as it makes reading the code easier.